internals/setup.js

```diff
@@ -0,0 +1,338 @@
+#!/usr/bin/env node
+/* eslint-disable import/no-extraneous-dependencies */
+
+const readPkg = require('read-pkg')
+const writePkg = require('write-pkg')
+const shell = require('shelljs')
+const inquirer = require('inquirer')
+const ora = require('ora')
+const rimraf = require('rimraf')
+const compareVersions = require('compare-versions')
+const chalk = require('chalk')
+const util = require('util')
+const exec = util.promisify(require('child_process').exec)
+
+const npmConfig = require('./helpers/get-npm-config')
+
+process.stdin.resume()
+process.stdin.setEncoding('utf8')
+
+process.stdout.write('\n')
+
+/**
+ * Deletes the current directory
+ *
+ * @returns {Promise<any>}
+ */
+function deleteCurrentDir() {
+  return new Promise((resolve, reject) => {
+    rimraf(__dirname, error => {
+      if (error) {
+        reject(new Error(error))
+      } else {
+        resolve()
+      }
+    })
+  })
+}
+
+/**
+ * Checks if we are under Git version control
+ * @returns {Promise<boolean>}
+ */
+async function hasGitRepository() {
+  const { stdout } = await exec('git status')
+  const regex = new RegExp(/fatal:\s+Not\s+a\s+git\s+repository/, 'i')
+  return !regex.test(stdout)
+}
+
+/**
+ * Checks if this is a clone from our repo
+ * @returns {Promise<any>}
+ */
+async function checkIfRepositoryIsAClone() {
+  const { stdout } = await exec('git remote -v')
+
+  const isClonedRepo = stdout
+    .split(/\r?\n/)
+    .map(line => line.trim())
+    .filter(line => line.startsWith('origin'))
+    .filter(line => /kefranabg\/bento-starter\.git/.test(line)).length
+
+  return !!isClonedRepo
+}
+
+/**
+ * Remove the current Git repository
+ * @returns {Promise<any>}
+ */
+function removeGitRepository() {
+  return new Promise((resolve, reject) => {
+    try {
+      shell.rm('-rf', '.git/')
+      resolve()
+    } catch (err) {
+      reject(err)
+    }
+  })
+}
+
+/**
+ * Ask user if he wants to start with a new repository
+ * @returns {Promise<boolean>}
+ */
+async function askUserIfWeShouldCreateNewRepo() {
+  const NEW_REPOSITORY = 'NEW_REPOSITORY'
+  const answers = await inquirer.prompt([
+    {
+      type: 'confirm',
+      message: 'Do you want to start with a new repository ?',
+      name: NEW_REPOSITORY
+    }
+  ])
+  return !!answers[NEW_REPOSITORY]
+}
+
+/**
+ * Ask user for new origin for this repository. If provided, call git remote add origin
+ * @returns {Promise<any>}
+ */
+async function askUserForNewRemote() {
+  const NEW_REMOTE = 'NEW_REMOTE'
+
+  const answers = await inquirer.prompt([
+    {
+      type: 'input',
+      message:
+        'Enter new remote origin (ex: https://github.com/user/my-repo.git) for this repository [Enter to cancel]:  ',
+      name: NEW_REMOTE
+    }
+  ])
+
+  const origin = answers[NEW_REMOTE]
+
+  if (origin) {
+    const spinner = ora('Adding new remote to repository')
```

Comment:
Oops ! I think `.start()` is missing

```suggestion
    const spinner = ora('Adding new remote to repository').start()
```